package.json

```diff
@@ -1975,30 +2035,77 @@
 					"command": "issue.copyGithubHeadLink",
 					"when": "github:hasGitHubRemotes",
 					"group": "1_githubPullRequests@3"
+				},
+				{
+					"command": "issue.copyGithubDevLinkFile",
+					"when": "github:hasGitHubRemotes && remoteName == codespaces || github:hasGitHubRemotes && embedderIdentifier == github.dev",
+					"group": "0_vscode@0"
 				}
 			],
-			"editor/title/context": [
+			"editor/lineNumber/context": [
 				{
 					"command": "issue.copyGithubPermalink",
-					"when": "github:hasGitHubRemotes",
-					"group": "1_cutcopypaste@10"
+					"when": "github:hasGitHubRemotes && activeEditor == workbench.editors.files.textFileEditor && config.editor.lineNumbers == on",
+					"group": "1_cutcopypaste@3"
+				},
+				{
+					"command": "issue.copyMarkdownGithubPermalink",
+					"when": "github:hasGitHubRemotes && activeEditor == workbench.editors.files.textFileEditor && config.editor.lineNumbers == on",
+					"group": "1_cutcopypaste@4"
 				},
 				{
 					"command": "issue.copyGithubHeadLink",
+					"when": "github:hasGitHubRemotes && activeEditor == workbench.editors.files.textFileEditor && config.editor.lineNumbers == on",
```

Comment:
Are all the `config.editor.lineNumbers == on` conditions needed? I would expect this to be controlled by VS Code core.

Reply:
I'm not sure it would make sense for core to handle hiding these commands, since the editor/lineNumber/context menu could also eventually include other extension-contributed items that we'd want to show regardless of line number enablement.

Reply:
This behavior also isn't final since I've already gotten feedback that it's unexpected that these commands would go away based on line numbers being toggled off to reduce visual noise.